### PR TITLE
add if else statement to show page so users can't add defintion when …

### LINKED ISSUE #65: fix add defintion bug
currently you have option to add definition when not logged in which breaks everything.

### PR DESCRIPTION
**Pull request by kevinpmcc**
…not signed in.

closes #65

### CODE REVIEW

**Comment by kevinpmcc** [comment]
this stops it breaking but not best user experience. 

when you click sign in on terms show page and sign in it doesn't bring you back to the terms show page of that term but back to the search page. 

for the moment code I've added stops it breaking but will need to fix the user flow issue later.

File: db/schema.rb [hunk 1]
```diff
@@ -90,10 +90,6 @@
 
   add_foreign_key "answers", "terms"
   add_foreign_key "answers", "users"
-  add_foreign_key "downvotes", "answers"
```
Comment (eripheebs) [inline db/schema.rb:93]: why are these deleted bb?

**Comment by kevinpmcc** [comment]
hmm weird, let me have a look and try again.

**Comment by kevinpmcc** [comment]
eugh there are more problems with this. came across below error when i was logged out and term had answer. I can't look at this today but if anyone has time please belt  away.

undefined method `has_upvoted?' for nil:NilClass

Upvotes: <%=answer.upvotes.count%>
        Downvotes: <%=answer.downvotes.count%>
        **<%if current_user.has_upvoted?(answer)%>**
          <button id="add-term-btn" class="btn btn-success" type="button">Upvote</button>
          <form role="form" action="/terms/<%=@term.id%>/answers/<%=answer.id%>/upvotes/<%=Upvote.find_by(user_id: current_user.id, answer_id: answer.id).id%>" method='delete'>
            <button id="add-term-btn" type="submit" class="btn btn-default">Downvote</button>

**Comment by eripheebs** [comment]
fixed above issue in different PR :) ^

**Comment by kevinpmcc** [comment]
can this be reviewed again as all seems to be working now.

**Comment by paulalexrees** [comment]
:shipit:
